src/tracer/MOM_tracer_advect.F90

```diff
@@ -125,12 +131,30 @@ subroutine advect_tracer(h_end, uhtr, vhtr, OBC, dt, G, GV, US, CS, Reg, x_first
   call cpu_clock_begin(id_clock_advect)
   x_first = (MOD(G%first_direction,2) == 0)
 
-  ! increase stencil size for Colella & Woodward PPM
-  use_PPM_stencil = CS%usePPM .and. .not. CS%useHuynhStencilBug
-  if (use_PPM_stencil) stencil = 3
+  ! Choose the maximum stencil from all the local advection scheme
+  do m = 1,ntr
+
+     local_advect_scheme(m) = Reg%Tr(m)%advect_scheme
+     if(local_advect_scheme(m) < 0) local_advect_scheme(m) = CS%default_advect_scheme
+
+     if (local_advect_scheme(m) == ADVECT_PLM) then
+       stencil_local = 2
+     elseif (local_advect_scheme(m) == ADVECT_PPM) then
+       stencil_local = 3
+     elseif (local_advect_scheme(m) == ADVECT_PPMH3) then
+       if (CS%useHuynhStencilBug) then
+         stencil_local = 2
+       else
+         stencil_local = 3
+       endif
+     endif
+     stencil = max(stencil, stencil_local)
+  enddo
 
-  ntr = Reg%ntr
-  Idt = 1.0 / dt
+  if (min(is-isd,ied-ie,js-jsd,jed-je).lt.stencil) then
```

Comment:
Although `.lt.` is legal Fortran, within the MOM6 code, we generally have a preference for the more succinct and clearer syntax ` < `.